src/parser/shaman/restoration/modules/talents/AncestralProtectionTotem.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+
+import fetchWcl from 'common/fetchWclApi';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import Icon from 'common/Icon';
+import { formatDuration } from 'common/format';
+import SPECS from 'game/SPECS';
+
+import LazyLoadStatisticBox, { STATISTIC_ORDER } from 'interface/others/LazyLoadStatisticBox';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+
+import Analyzer from 'parser/core/Analyzer';
+import Combatants from 'parser/shared/modules/Combatants';
+
+//RQXjBJ1kG9pAC2DV/21-Mythic++Atal'Dazar+-+Kill+(51:52)/Koorshaman/
+
+class AncestralProtectionTotem extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  loaded = false;
+  aptEvents = [];
+  constructor(...args) {
+    super(...args);
+    this.active = !!this.selectedCombatant.hasTalent(SPELLS.ANCESTRAL_PROTECTION_TOTEM_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+  }
+
+  // recursively fetch events until no nextPageTimestamp is returned
+  fetchAll(pathname, query) {
+    const checkAndFetch = async _query => {
+      const json = await fetchWcl(pathname, _query);
+      Array.prototype.push.apply(this.aptEvents, json.events);
+      if (json.nextPageTimestamp) {
+        return checkAndFetch(Object.assign(query, {
+          start: json.nextPageTimestamp,
+        }));
+      }
+      this.loaded = true;
+      return null;
+    };
+    return checkAndFetch(query);
+  }
+
+  load() {
+    const query = {
+      start: this.owner.fight.start_time,
+      end: this.owner.fight.end_time,
+      filter: `(
+        type='cast'
```

Comment:
Need to add braces around the AND statements to ensure they’re parses correctly. Right now operator priority is ambiguous, especially since it is undefined in any documentation